lib/publiccloud/ec2.pm

```diff
@@ -17,13 +17,58 @@ use testapi;
 
 has ssh_key      => undef;
 has ssh_key_file => undef;
+has credentials  => undef;
+
+sub create_credentials {
+    my ($self) = @_;
+
+    if (!defined($self->credentials)) {
+        my $credentials = $self->do_rest_api('/ec2/users', method => 'post');
+        die("Missing key") if (!defined($credentials->{keys}));
+        for my $key (@{$credentials->{keys}}) {
+            next if (!defined($key->{secret}));
```

Comment:
hm and why service will return key which is not active or without a password ? and why it will return a list while we need only one and don't care which ? 
would like to have service call which return single key/value pair and doing this loops and check on other end